src/app/store/modules/river-events.js

```diff
@@ -1,45 +1,135 @@
-import actions from '@/app/store/actions'
-import mutations from '@/app/store/mutations'
-import { getReachEvents } from '@/app/services'
+import actions from "@/app/store/actions";
+import mutations from "@/app/store/mutations";
+import { getReachEvents } from "@/app/services";
+import { cloneDeep } from "lodash";
 
 export default {
   namespaced: true,
   state: {
     error: false,
     loading: false,
-    data: null
+    data: null,
   },
   mutations,
   actions: {
     ...actions,
     async getProperty(context, id) {
       try {
-        context.commit('DATA_REQUEST')
-        
-        const result = await getReachEvents(id)
+        context.commit("DATA_REQUEST");
+
+        const result = await getReachEvents(id);
 
         if (!result.errors) {
-          context.commit('DATA_SUCCESS', result)
+          context.commit("DATA_SUCCESS", result);
         }
-
       } catch (error) {
-        context.commit('DATA_ERROR', error)
+        context.commit("DATA_ERROR", error);
       }
-    }
+    },
   },
   getters: {
-    releases: state => {
-      if (state.data) {
-        const results = state.data.filter(item => !!item.data && item.data.length && item.data[0]?.category)
+    // this will return all event dates with the parent event tucked into "event", sorted descending by date.
+    releaseDates: (state, getters) => {
+      let intermediate = [];
+
+      if (getters.releaseEvents?.length) {
+        getters.releaseEvents.forEach((re) => {
+          intermediate = intermediate.concat(
+            re.dates.map((y) => ({ ...y, event: re }))
+          );
+        });
+
+        return cloneDeep(
+          intermediate.sort((a, b) => b.event_date.localeCompare(a.event_date))
+        );
+      }
+      return [];
+    },
+
+    /**
+     * Return the next release date, from today or last release that was listed.
+     * @param state
+     * @param getters
+     * @return {null}
+     */
+    closestReleaseDate: (state, getters) => {
+      if (getters.releaseDates?.length) {
+        const todayDate = new Date();
+        todayDate.setHours(0, 0, 0);
+
+        let today = todayDate.toISOString();
+        return getters.releaseDates.find((x) => x.event_date < today) ?? null;
+      }
+      return null;
+    },
+    // puts today at the top, then counts forward, then counts backward from today
+    // probably how most people would use this is "when is the next release?" at the top,
+    // "when are the next couple releases?" near the top, "when were was the last release?"
+    // scroll down
+    releasesFromTodayThenPast: (state, getters) => {
```

Comment:
I don't think @AWKevin asked for this in the task but maybe I'm missing something. It strikes me as a confusing way to represent the data